vllm/hpu/ops.py

```diff
@@ -12,86 +12,49 @@
 from typing import List, Optional, Tuple
 
 def silu_and_mul(output, input):
-    htorch.core.mark_step()
     d = input.shape[-1] // 2
     silu = torch.nn.SiLU().to(input.device)
     x, y = torch.split(input, d, dim=-1)
     output.copy_(silu(x) * y)
-    htorch.core.mark_step()
+
 
 def gelu_new(output, input):
     raise NotImplementedError
 
+
 def gelu_fast(output, input):
     raise NotImplementedError
 
+
+def fetch_from_cache(cache, blocks):
+    _, seq_len = blocks.shape
+    return torch.cat([cache.index_select(0, blocks[:, i]) for i in range(seq_len)], dim=-1).flatten(0, 1)
+
+
+
 def paged_attention_v1(query_in, key_cache_in, value_cache_in, head_mapping, scale, block_tables, context_lens, block_size, max_context_len, alibi_slopes, attn_masks=None)  -> None:
```

Comment:
The `_in` suffix is meant to distinguish inputs from the tensors used in the implementation, which are cast to BF16 to ensure the proper functioning of the filler in softmax. Does this version of PA deliver good results on FP16? If no, the type casting may need to be restored. If yes, then the suffixes should be removed.

Reply:
I've double check and it works fine both on fp16 and bf16. I'll remove those suffixes.